### PR TITLE
Feature/tiffstack

### PR DESCRIPTION
**Pull request by petersteneteg**
rebase of #413

### CODE REVIEW

File: modules/base/src/properties/basisproperty.cpp [hunk 1]
```diff
@@ -37,186 +39,267 @@ std::string BasisProperty::getClassIdentifier() const { return classIdentifier;
...
+}
+
+BasisProperty& BasisProperty::operator=(const BasisProperty& that) {
+    if (this != &that) {
+        CompositeProperty::operator=(that);
+        mode_ = that.mode_;
+        reference_ = that.reference_;
+        overRideDefaults_ = that.overRideDefaults_;
+        size_ = that.size_;
+        a_ = that.a_;
```
Comment (ResearchDaniel) [inline modules/base/src/properties/basisproperty.cpp:140]: Is this not taken care of by CompositeProperty= operator?

File: tests/volumes/boronslice/generate.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#Inviwo Python script 
+import inviwopy
+
+
+app = inviwopy.app
+network = app.network
+
+base = "C:/Users/petst55.AD/Documents/Inviwo/inviwo/tests/volumes/boronslice/boron{:03}.png"
+
```
Comment (ResearchDaniel) [inline tests/volumes/boronslice/generate.py:9]: Relative path

File: include/inviwo/core/io/datareaderfactory.h [hunk 3]
```diff
@@ -59,6 +59,14 @@ class IVW_CORE_API DataReaderFactory : public Factory<DataReader, const FileExte
     std::unique_ptr<DataReaderType<T>> getReaderForTypeAndExtension(const std::string& ext);
     template <typename T>
     std::unique_ptr<DataReaderType<T>> getReaderForTypeAndExtension(const FileExtension& ext);
+    template <typename T>
+    std::unique_ptr<DataReaderType<T>> getReaderForTypeAndExtension(const FileExtension& ext,
+                                                                    const std::string& fallbackExt);
```
Comment (ResearchDaniel) [inline include/inviwo/core/io/datareaderfactory.h:71]: Second arg should be FileExtension type?
Could also mention its use case

**Comment by ResearchDaniel** [comment]
A bit hard to see if everything works but looks good. Some minor comments

**Comment by petersteneteg** [comment]
Comments?

**Review by ResearchDaniel — CHANGES_REQUESTED** [review]

File: modules/base/src/properties/basisproperty.cpp (hunk 1 above)
Comment (petersteneteg) [reply modules/base/src/properties/basisproperty.cpp:140]: Nopes

File: tests/volumes/boronslice/generate.py (hunk 2 above)
Comment (petersteneteg) [reply tests/volumes/boronslice/generate.py:9]: this is run from the python editor, have no idea what a relative path should be.. one has to edit it before use

File: include/inviwo/core/io/datareaderfactory.h (hunk 3 above)
Comment (petersteneteg) [reply include/inviwo/core/io/datareaderfactory.h:71]: the second arg has to be a string, since we don't have any description for the file extension...

**Review by ResearchDaniel — APPROVED** [review]

**Pull request merged**